hal/pinmap.h

```diff
@@ -43,6 +43,12 @@ typedef struct {
     const int *peripheral;
 } PeripheralList;
 
+typedef struct {
+    int peripheral;    // Explicitly specified peripheral
+    PinName *pin;;     // Pointer to array which contains pins
+    int *function;     // Pointer to array which contains pins functions
```

Comment:
These two should be const-qualified? Shouldn't be being modified via the pinmap.